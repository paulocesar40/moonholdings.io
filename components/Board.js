import React from 'react';
import styled from 'styled-components';

import Welcome from './Welcome';
import Astronaut from './Astronaut';

const StyledBoard = styled.div`
  display: flex;
  justify-content: center;
  margin: 0;
  padding: 0;
  width: 100%;
  height: 100%;
`;

class Board extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      portfolio: [],
      loading: true
    };
  }

  render() {
    const { portfolio } = this.state;
    const hasPortfolio = portfolio.length > 0;

    return (
      <StyledBoard>
        <Welcome />
        {/* <PlusButton toggleSearch={this.handleSearchButton} /> */}
        {/* <Affiliates /> */}
        {/* <Nomics /> */}
        <Astronaut showLogo={hasPortfolio} />
      </StyledBoard>
    );
  }
}

export default Board;
