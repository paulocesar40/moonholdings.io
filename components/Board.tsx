import React from 'react';
import styled from 'styled-components';

import Welcome from './Welcome';
import Astronaut from './Astronaut';
import NomicsLink from './NomicsLink';

interface IState {
  portfolio: [];
  loading: boolean;
}

const StyledBoard = styled.div`
  display: flex;
  justify-content: center;
  margin: 0;
  padding: 0;
  width: 100%;
  height: 100%;
`;

class Board extends React.Component<{}, IState> {
  constructor(props: IState) {
    super(props);

    this.state = {
      portfolio: [],
      loading: true
    };
  }

  render() {
    const { portfolio } = this.state;
    const hasPortfolio = portfolio.length > 0;

    return (
      <StyledBoard>
        <Welcome />
        {/* <PlusButton toggleSearch={this.handleSearchButton} /> */}
        {/* <Affiliates /> */}
        <NomicsLink />
        <Astronaut showLogo={hasPortfolio} />
      </StyledBoard>
    );
  }
}

export default Board;
