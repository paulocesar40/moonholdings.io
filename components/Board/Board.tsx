import React from 'react'
import { bind } from 'decko'

import { Welcome, Astronaut, NomicsLink, PlusButton, Portfolio } from '../'
import { IPortfolioItem } from '../../shared/types'
import { StyledBoard } from '../../styles'

interface IState {
  portfolio: IPortfolioItem[];
  loading: boolean;
}

//@TODO to remove...
const tempPortfolio = [
  {
    balance: 2.46781018,
    symbol: 'BTC',
    marketCap: 63636922279.28325,
    name: 'Bitcoin',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'DCR',
    marketCap: 63636922279.28325,
    name: 'Decred',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'ETH',
    marketCap: 63636922279.28325,
    name: 'Ethereum',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'BNB',
    marketCap: 63636922279.28325,
    name: 'Binance',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'LTC',
    marketCap: 63636922279.28325,
    name: 'Litecoin',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'LSK',
    marketCap: 63636922279.28325,
    name: 'Lisk',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },{
    balance: 2.46781018,
    symbol: 'ZRX',
    marketCap: 63636922279.28325,
    name: '0xProject',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'MKR',
    marketCap: 63636922279.28325,
    name: 'Maker',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'USDT',
    marketCap: 63636922279.28325,
    name: 'Tether',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  },
  {
    balance: 2.46781018,
    symbol: 'NANO',
    marketCap: 63636922279.28325,
    name: 'Nano',
    percentage: 66.4,
    price: 3637.33607,
    value: 8976.25,
  }
]

class Board extends React.Component<{}, IState> {
  constructor(props: IState) {
    super(props);

    this.state = {
      portfolio: tempPortfolio,
      loading: true
    };
  }

  render() {
    const { portfolio } = this.state;
    const hasPortfolio = portfolio.length > 0;

    return (
      <StyledBoard>
        { !hasPortfolio && <Welcome /> }
        <Portfolio
          coins={portfolio}
          edit={this.toggleSquareEdit}
        />
        <PlusButton toggleSearch={this.handleOnSearch} />
        <NomicsLink />
        <Astronaut showLogo={hasPortfolio} />
      </StyledBoard>
    );
  }

  @bind
  private handleOnSearch() {
    console.log('handleSearchButton...');
  }

  @bind
  private toggleSquareEdit(toggle: boolean, coin: IPortfolioItem) {
    console.log('toggleSquareEdit...', coin);
  }
}

export default Board;
