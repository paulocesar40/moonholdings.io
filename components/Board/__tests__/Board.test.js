import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Board from '../Board'
import { Welcome, Astronaut, NomicsLink, PlusButton } from '../../'

describe('<Astronaut /> component', () => {
  describe('rendering', () => {
    const wrapper = shallow(<Board />);
    it('should render a component matching the snapshot', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
      expect(wrapper).toHaveLength(1);
      expect(wrapper.contains(<Welcome/>));
      expect(wrapper.contains(<Astronaut showLogo={true}/>));
      expect(wrapper.contains(<NomicsLink/>));
      expect(wrapper.contains(<PlusButton toggleSearch={jest.fn()}/>));
    });
  });
});
