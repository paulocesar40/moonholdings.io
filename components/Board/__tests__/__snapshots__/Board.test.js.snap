// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Astronaut /> component rendering should render a component matching the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
